src/exberry_int/integration_exberry.py

```diff
@@ -311,12 +323,20 @@ def create_order(order_data):
                 'price': float(order_data['price']),
                 'side': order_data['side'],
                 'timeInForce': order_data['timeInForce'],
+                'expiryDate': order_data['expiryDate'],
                 'mpOrderId': order_data['mpOrderId'],
                 'userId': order_data['userId'],
             },
             'q': EXBERRY_PLACE_ORDER,
             'sid': order_data['mpOrderId']
         }
+        # Market orders cannot include a price on their order submission
+        if order_data['orderType'] == 'Market':
+            order_json['d'].pop('price')
+        # Only GTD (Good Till Date) orders can submit an expirydate on their order
+        if order_data['timeInForce'] != 'GTD':
+            order_json['d'].pop('expiryDate')
```

Comment:
Oh weird... if I issue an order with these parameters through the UI, I wonder what values get populated in these fields? 

I would also argue that this validation should get moved into the Daml side. I.e., mark these fields as Optional and assert None for these cases...

Reply:
> Oh weird... if I issue an order with these parameters through the UI, I wonder what values get populated in these fields?

You can't provide these values via the UI of the Marketplace as the Marketplace order model doesn't allow for it. These values are set when converting to the Exberry `NewOrderRequest` in the Marketplace bot. Ie, if the order doesn't contain these values, then they get defaulted to `-1` as its a mandatory field in the Exberry Integration daml and the check above ensure's it's removed from the JSON before its sent to Exberry. 

> I would also argue that this validation should get moved into the Daml side. I.e., mark these fields as Optional and assert None for these cases...

When I was writing this code, I couldn't find any documentation or example on how Dazl handle's optional fields and thus could only think of this dirty workaround. If you know how to do it or know where I can find this information, let me know and I can update it.